src/interpret.c

```diff
@@ -6758,10 +6761,40 @@ check_function_args(int fx, program_t *progp, bytecode_p funstart)
                     // or it is a left-over from last execution.
                     inter_pc = funstart;
                 }
-                // unravel any lvalue indirection (if any)
-                errorf("Bad arg %d to %s(): got '%s', expected '%s'.\n"
-                       , i+1, get_txt(header->name), buff,
-                       get_lpctype_name(arg_type[i]));
+
+                if (progp->flags & P_WARN_RTT_CHECKS)
+                {
+                    // We need the stack frames to continue execution.
+                    // But now they are in the way of a good error message...
+                    struct control_stack saved_csf[3];
+
+                    push_control_stack(inter_sp, inter_pc, inter_fp, inter_context);
+                    for (int j = 0; j <= num_csf; j++)
```

Comment:
Mhmm, würde mich wohler fühlen, hier auch explizit sicherzustellen, dass num_cfg/j < 3 bleibt. Nicht dass wir irgendwann an den control frames rumbasteln, oben ggf. auch mal einen mehr entfernen, aber nicht mehr dran denken, dass die hier noch in nem begrenztem Array gesichert werden.